cmake/AwsSIMD.cmake

```
@@ -35,7 +46,22 @@ if (USE_CPU_EXTENSIONS)
             _mm256_permutevar8x32_epi32(vec, vec);
 
             return 0;
-        }"  HAVE_AVX2_INTRINSICS)
+        }"  AWS_HAVE_AVX2_INTRINSICS)
+
+    check_c_source_compiles("
+        #include <nmintrin.h>
+        int main() {
+            __m128i a = _mm_setzero_si128();
+            return 0;
+        }" AWS_HAVE_SSE42_INTRINSICS)
+
+    check_c_source_compiles("
+        #include <immintrin.h>
+
+        int main() {
+            __m512 a = _mm512_setzero_ps();
+            return 0;
+        }" AWS_HAVE_AVX512_INTRINSICS)
```

debatable: you could speed up configuration by skipping these `check_c_source_compiles()` checks when you know they're going to fail

```suggestion
    if (HAVE_M_AVX512_FLAG)
        check_c_source_compiles("
            #include <immintrin.h>

            int main() {
                __m512 a = _mm512_setzero_ps();
                return 0;
            }" AWS_HAVE_AVX512_INTRINSICS)
    endif()
```

---

sure.

---

i think in hindsight, I don't want to make that coupling, one is for figuring out flags, the other is for checking the compiler supports the code. If those are ever out of sync I'd like it to get caught right here.